src/backend/tests/it/utils/pocketic.rs

```diff
@@ -309,3 +313,94 @@ where
         WasmResult::Reject(error) => Err(error),
     })
 }
+
+/// A test Oisy backend canister with a shared reference to the `PocketIc` instance it is installed on.
+pub struct PicBackend {
+    pub pic: Arc<PocketIc>,
+    pub canister_id: Principal,
+}
+impl PicCanisterTrait for PicBackend {
+    fn pic(&self) -> Arc<PocketIc> {
+        self.pic.clone()
+    }
+    fn canister_id(&self) -> Principal {
+        self.canister_id.clone()
+    }
+}
+impl PicBackend {
+    /// Creates toy users with the given range of principals.
+    pub fn create_users(&self, range: Range<u8>) -> Vec<OisyUser> {
+        let mut expected_users: Vec<OisyUser> = Vec::new();
+        for i in range {
+            self.pic.advance_time(Duration::new(10, 0));
+            let caller = Principal::self_authenticating(i.to_string());
+            let response = self.update::<UserProfile>(caller, "create_user_profile", ());
+            let timestamp = self.pic.get_time();
+            let timestamp_nanos = timestamp
+                .duration_since(UNIX_EPOCH)
+                .expect("Time went backwards")
+                .as_nanos();
+            let expected_user = OisyUser {
+                updated_timestamp: timestamp_nanos as u64,
+                pouh_verified: false,
+                principal: caller,
+            };
+            expected_users.push(expected_user);
+            assert!(response.is_ok());
+        }
+        expected_users
+    }
+}
+
+/// Common methods for interacting with a canister using `PocketIc`.
+pub trait PicCanisterTrait {
+    /// A shared PocketIc instance.
+    ///
+    /// Note: `PocketIc` uses interior mutability for query and update calls.  No external mut annotation or locks appear to be necessary.
+    fn pic(&self) -> Arc<PocketIc>;
+
+    /// The ID of this canister.
+    fn canister_id(&self) -> Principal;
+
+    /// Makes an update call to the canister.
```

Comment:
The query and update calls here are copied (and adjusted) from https://github.com/dfinity/oisy-wallet/blob/cdf4246f4943215deb6eb40c16bf8cabdf5ee8e4/src/backend/tests/it/utils/pocketic.rs#L259